src/main/java/sg/ncl/MainController.java

```diff
@@ -1918,6 +1948,94 @@ public String validateExperiment(
         return "redirect:/experiments";
     }
 
+    @RequestMapping(value = "/experiments/save_image/{teamId}/{expId}/{nodeId}", method = RequestMethod.GET)
+    public String saveExperimentImage(@PathVariable String teamId, @PathVariable String expId, @PathVariable String nodeId, Model model) {
+        Map<String, Map<String, String>> singleNodeInfoMap = new HashMap<>();
+        Image saveImageForm = new Image();
+
+        String teamName = invokeAndExtractTeamInfo(teamId).getName();
+        Realization realization = invokeAndExtractRealization(teamName, Long.parseLong(expId));
+
+        // experiment may have many nodes
+        // extract just the particular node details to display
+        for (Map.Entry<String, Map<String, String>> nodesInfo : realization.getNodesInfoMap().entrySet()) {
+            String nodeName = nodesInfo.getKey();
+            Map<String, String> singleNodeDetailsMap = nodesInfo.getValue();
+            if (singleNodeDetailsMap.get("nodeId").equals(nodeId)) {
+                singleNodeInfoMap.put(nodeName, singleNodeDetailsMap);
+                // store the current os of the node into the form also
+                // have to pass the the services
+                saveImageForm.setCurrentOS(singleNodeDetailsMap.get("os"));
+            }
+        }
+
+        saveImageForm.setTeamId(teamId);
+        saveImageForm.setNodeId(nodeId);
+
+        model.addAttribute("teamName", teamName);
+        model.addAttribute("singleNodeInfoMap", singleNodeInfoMap);
+        model.addAttribute("pathTeamId", teamId);
+        model.addAttribute("pathExperimentId", expId);
+        model.addAttribute("pathNodeId", nodeId);
+        model.addAttribute("experimentName", realization.getExperimentName());
+        model.addAttribute("saveImageForm", saveImageForm);
+        return "save_experiment_image";
+    }
+
+    // bindingResult is required in the method signature to perform the JSR303 validation for Image object
+    @RequestMapping(value = "/experiments/save_image/{teamId}/{expId}/{nodeId}", method = RequestMethod.POST)
+    public String saveExperimentImage(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: sg.ncl.exceptions.WebServiceRuntimeException, com.fasterxml.jackson.core.JsonProcessingException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1160)